transmart-core-db/src/main/groovy/org/transmartproject/db/multidimquery/DataTableImpl.groovy

```diff
@@ -0,0 +1,219 @@
+package org.transmartproject.db.multidimquery
+
+import com.google.common.collect.AbstractIterator
+import com.google.common.collect.HashBasedTable
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.google.common.collect.Lists
+import com.google.common.collect.PeekingIterator
+import com.google.common.collect.Table
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.Immutable
+import groovy.transform.ToString
+import org.transmartproject.core.dataquery.SortOrder
+import org.transmartproject.core.multidimquery.DataTable
+import org.transmartproject.core.multidimquery.DataTableColumn
+import org.transmartproject.core.multidimquery.DataTableRow
+import org.transmartproject.core.multidimquery.Dimension
+import org.transmartproject.core.multidimquery.Hypercube
+import org.transmartproject.core.multidimquery.HypercubeValue
+
+import static java.util.Objects.requireNonNull
+
+@CompileStatic
+class DataTableImpl implements DataTable {
```

Comment:
Looks like a reimplementation of HypercubeTabularResultView. Any idea on how to bring those two implementations together?